runtime/lua/vim/_watch.lua

```diff
@@ -0,0 +1,170 @@
+-- Create a file watcher, each watcher is identified by the buffer number of the file it is watching.
```

Comment:
@BK1603 Do you foresee this eventually evolving into something more general-purpose to use for things beyond autoread?

I'd be interested in leveraging something like this to drive LSP `workspace/didChangeWatchedFiles` notifications in #21293. Right now that PR tightly couples the LSP and file watching details, so a slightly higher-level file watching interface that this might provide would help clean that up a lot.

I totally understand those kinds of changes are out of scope for this particular PR, but just a heads-up that I might be proposing some changes here after this PR lands.

Reply:
@nojnhuh As a user who once tried very hard to make Neovim work with Unity, I very much appreciate your linked PR! Thank you!